### PR TITLE
feat: changes on Embed APIs

### PR DESCRIPTION
**Pull request by tl-genie**
## Description

- Added new Embed APIs
- Applied changes regarding to Embed APIs

## Type of Change

- [x] New feature (non-breaking change which adds functionality)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation update
- [ ] Others

## Checklist

Before you submit this PR, please make sure you have completed the following:

- [x] I have read the [CONTRIBUTING.md](https://github.com/twelvelabs-io/twelvelabs-python/blob/main/CONTRIBUTE.md) document.
- [x] I have ensured my PR title is descriptive and in imperative mood.
- [x] I have added necessary documentation (if appropriate).

## Further comments

Example code runs fine.

image
image
image

### CODE REVIEW

**Review by henry-hs-choi — APPROVED** [review]
LGTM 😄

**Review by henry-hs-choi — APPROVED** [review]

File: examples/embed.py [hunk 1]
```diff
@@ -11,30 +11,56 @@
...
         text_truncate="start",
     )
-    print(f"Created embedding: engine_name={embedding.engine_name}")
+    print(f"Created text embedding: engine_name={text_embedding.engine_name}")
 
     video_path = os.path.join(os.path.dirname(__file__), "assets/example.mp4")
     task = client.embed.task.create(engine_name=engine_name, video_file=video_path)
     print(
-        f"Created task: id={task.id} engine_name={task.engine_name} status={task.status}"
+        f"Created task of text embedding: id={task.id} engine_name={task.engine_name} status={task.status}"
```
Comment (park-yeonhoo) [inline examples/embed.py:46]:
```suggestion
        f"Created task: id={task.id} engine_name={task.engine_name} status={task.status}"
```

Comment (park-yeonhoo) [reply examples/embed.py:46]: task is used for video embedding

File: examples/embed.py [hunk 2]
```diff
@@ -11,30 +11,56 @@
...
+    print(f"Created image embedding: engine_name={image_embedding.engine_name}")
+
+    task = client.embed.task.create(engine_name=engine_name, video_file=video_path)
+    print(
+        f"Created task of image embedding: id={task.id} engine_name={task.engine_name} status={task.status}"
+    )
+
+    status = task.wait_for_done(callback=on_task_update)
+    print(f"Embedding done: {status}")
+    print_video_embeddings(task.id)
```
Comment (park-yeonhoo) [inline examples/embed.py:66]: duplicate. can we rather have an example of audio embedding?

File: examples/embed.py [hunk 3]
```diff
@@ -11,30 +11,56 @@
...
     engine_name = "Marengo-retrieval-2.6"
-    embedding = client.embed.create(
+
+    text_embedding = client.embed.create(
         engine_name=engine_name,
         text="man walking across the street",
         text_truncate="start",
     )
-    print(f"Created embedding: engine_name={embedding.engine_name}")
+    print(f"Created text embedding: engine_name={text_embedding.engine_name}")
```
Comment (andreipope) [inline examples/embed.py:41]: I think this should print the embedding itself.

File: examples/embed.py (hunk 1 above)
Comment (tl-genie) [reply examples/embed.py]: plz check it again!

File: examples/embed.py (hunk 3 above)
Comment (tl-genie) [reply examples/embed.py]: that'd be great :)

**Review by park-yeonhoo — APPROVED** [review]

**Review by park-yeonhoo — APPROVED** [review]

File: twelvelabs/models/embed.py [hunk 4]
```diff
@@ -50,37 +50,29 @@ class EmbeddingMediaMetadata(BaseModel):
...
     error_message: Optional[str] = None
     metadata: Optional[EmbeddingMediaMetadata] = None
 
 
 class SegmentEmbedding(BaseModel):
-    start_offset_sec: float
+    start_offset_sec: Optional[float] = None
     end_offset_sec: Optional[float] = None
     embedding_scope: Optional[str] = None
     values: Optional[List[float]] = Field(default=None, alias="float")
```
[the remark this answers is not in the document]
Comment (tl-genie) [reply twelvelabs/models/embed.py]: applied!

File: twelvelabs/models/embed.py [hunk 5]
```diff
@@ -59,7 +59,7 @@ class SegmentEmbedding(BaseModel):
     start_offset_sec: Optional[float] = None
     end_offset_sec: Optional[float] = None
     embedding_scope: Optional[str] = None
-    values: Optional[List[float]] = Field(default=None, alias="float")
+    embedding_float: Optional[List[float]] = Field(default=None, alias="float")
```
Comment (park-yeonhoo) [inline twelvelabs/models/embed.py:62]:
```suggestion
    embeddings_float: Optional[List[float]] = Field(default=None, alias="float")
```

**Review by park-yeonhoo — APPROVED** [review]

**Pull request merged**